prow/config/config.go

```diff
@@ -286,44 +286,53 @@ func Load(prowConfig, jobConfig string) (*Config, error) {
 		return nil, err
 	}
 
-	if !stat.IsDir() {
-		// still support a single file
-		jobConfig, err := yamlToConfig(jobConfig)
+	if stat.IsDir() {
+		return nil, fmt.Errorf("jobConfig should be a single file!")
+	}
+
+	// split into multiple blobs :-)
+	jobBlobs, err := fileToYamlBlobs(jobConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error parse %s: %v", jobConfig, err)
+	}
+
+	for _, jobBlob := range jobBlobs {
+		subConfig, err := blobToConfig(jobBlob)
 		if err != nil {
 			return nil, err
 		}
-		if err := nc.mergeJobConfig(jobConfig.JobConfig); err != nil {
+		if err = nc.mergeJobConfig(subConfig.JobConfig); err != nil {
 			return nil, err
 		}
-		return nc, nil
 	}
+	return nc, nil
+}
 
-	err = filepath.Walk(jobConfig, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			logrus.WithError(err).Errorf("walking path %q.", path)
-			// bad file should not stop us from parsing the directory
-			return nil
-		}
-
-		if filepath.Ext(path) != ".yaml" {
-			return nil
-		}
-
-		if info.IsDir() {
-			return nil
-		}
+// fileToYamlBlobs convert one or more multiple yaml documents in a yaml file into a slice of yaml blobs
+func fileToYamlBlobs(yamlPath string) ([][]byte, error) {
+	file, err := os.Open(yamlPath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
 
-		subConfig, err := yamlToConfig(path)
-		if err != nil {
-			return err
+	splitRegexp := regexp.MustCompile("^-{3,}$")
```

Comment:
I would be surprised if the YAML unmarshaller doesn't already do this -- the base Golang JSON unmarshaller when taking a stream will give you every JSON document in succession

Reply:
hummm is it? See https://github.com/kubernetes/test-infra/pull/8313, I saw the streaming decoder is added not long ago, and seems we don't want to bump our yaml library.